### PR TITLE
Use encrypted user defaults instead of keychain for auth history

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Use encrypted user defaults instead of keychain for auth history

### CODE REVIEW

**Comment by clpolet** [comment]
Build is failing on tests, need some to fix the errors from the CI.

File: AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m [hunk 1]
```diff
@@ -209,25 +219,30 @@ - (void)removeAuthToken:(nullable NSString *)authToken {
...
-    self.authTokenHistoryArray = authTokenHistory;
+  self.authTokenHistoryArray = authTokenHistory;
+  NSData *decryptedData = authTokenHistory ? [NSKeyedArchiver archivedDataWithRootObject:(id)authTokenHistory] : nil;
+  NSData *encryptedData = decryptedData ? [self.encrypter encryptData:decryptedData] : nil;
+  [MS_USER_DEFAULTS setObject:encryptedData forKey:kMSAuthTokenHistoryKey];
+  if (encryptedData) {
+    MSLogDebug([MSAppCenter logTag], @"Saved new history state.");
   } else {
-    MSLogWarning([MSAppCenter logTag], @"Failed to save new history state in the keychain.");
+    MSLogWarning([MSAppCenter logTag], @"Failed to saved new history state.");
```
Comment (annakocheshkova) [inline AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m]:
```suggestion
    MSLogWarning([MSAppCenter logTag], @"Failed to save new history state.");
```

File: AppCenter/AppCenter/Internals/Util/MSEncrypter.m [hunk 2]
```diff
@@ -76,61 +76,70 @@ - (void)dealloc {
 }
 
 - (NSString *_Nullable)encryptString:(NSString *)string {
+  NSData *dataToEncrypt = [string dataUsingEncoding:NSUTF8StringEncoding];
+  return [[self encryptData:dataToEncrypt] base64EncodedStringWithOptions:0];
```
Comment (annakocheshkova) [inline AppCenter/AppCenter/Internals/Util/MSEncrypter.m]: Please use a constant https://developer.apple.com/documentation/foundation/nsdatabase64encodingoptions?language=objc

File: AppCenter/AppCenterTests/MSAuthTokenContextTests.m [hunk 3]
```diff
@@ -396,10 +393,9 @@ - (void)testCheckIfTokenNeedsToBeRefreshed {
   [self.sut addDelegate:delegateMock];
   OCMReject([delegateMock authTokenContext:OCMOCK_ANY refreshAuthTokenForAccountId:OCMOCK_ANY]);
 
-  NSArray<MSAuthTokenValidityInfo *> *mockArray = @[ [[MSAuthTokenValidityInfo alloc] initWithAuthToken:expectedAuthToken
-                                                                                              startTime:nil
-                                                                                                endTime:nil] ];
-  OCMStub([MSMockKeychainUtil arrayForKey:kMSAuthTokenHistoryKey]).andReturn(mockArray);
+  NSArray<MSAuthTokenInfo *> *authTokenHistory = @[ [[MSAuthTokenInfo alloc] initWithAuthToken:expectedAuthToken
+                                                                                              accountId:@"test" startTime:nil expiresOn:nil] ];
```
Comment (annakocheshkova) [inline AppCenter/AppCenterTests/MSAuthTokenContextTests.m]:
```suggestion
                                                                                              accountId:@"test" startTime:nil expiresOn:nil]];
```

File: AppCenter/AppCenterTests/MSAuthTokenContextTests.m [hunk 4]
```diff
@@ -396,10 +393,9 @@ - (void)testCheckIfTokenNeedsToBeRefreshed {
   [self.sut addDelegate:delegateMock];
   OCMReject([delegateMock authTokenContext:OCMOCK_ANY refreshAuthTokenForAccountId:OCMOCK_ANY]);
 
-  NSArray<MSAuthTokenValidityInfo *> *mockArray = @[ [[MSAuthTokenValidityInfo alloc] initWithAuthToken:expectedAuthToken
-                                                                                              startTime:nil
-                                                                                                endTime:nil] ];
-  OCMStub([MSMockKeychainUtil arrayForKey:kMSAuthTokenHistoryKey]).andReturn(mockArray);
+  NSArray<MSAuthTokenInfo *> *authTokenHistory = @[ [[MSAuthTokenInfo alloc] initWithAuthToken:expectedAuthToken
```
Comment (annakocheshkova) [inline AppCenter/AppCenterTests/MSAuthTokenContextTests.m]:
```suggestion
  NSArray<MSAuthTokenInfo *> *authTokenHistory = @[[[MSAuthTokenInfo alloc] initWithAuthToken:expectedAuthToken
```

**Comment by MatkovIvan** [comment]
@clpolet 

```
Failing tests:
	-[MSDistributeTests testDontShowConfirmationAlertIfNoMandatoryReleaseWhileNoNetwork]
```

That's not related to these changes. Need to be fixed separately.

File: AppCenter/AppCenterTests/MSAuthTokenContextTests.m (hunk 3 above)
Comment (MatkovIvan) [reply AppCenter/AppCenterTests/MSAuthTokenContextTests.m]: That's from formatter

File: AppCenter/AppCenterTests/MSAuthTokenContextTests.m (hunk 4 above)
Comment (MatkovIvan) [reply AppCenter/AppCenterTests/MSAuthTokenContextTests.m]: That's from formatter

File: AppCenter/AppCenter/Internals/Util/MSEncrypter.m (hunk 2 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/Internals/Util/MSEncrypter.m]: There are no options needed + I don't want to change the behavior here

**Pull request merged**